### ISSUE TITLE
Database Support

### ISSUE THREAD

**Issue by MR-Alex42** [Frooxius/NeosPublic#240]
For commercial projects there will likely be the need to have some kind of database support inside of Neos. Currently there is no support for higher level data handling next to C# coding.

**Comment by Frooxius** [comment]
I'm not sure what does this mean specifically? In what form do you want the database support to be? Would this be for making database connectors available to LogiX? Or some components representing a database connection that you can do queries against? 

Would this be generalized or made for specific types of databases?

It might be more feasible to simply extend the web nodes to support all common HTTP verbs with appropriate bodies and communicate with databases over some REST API. Then adding support for particular databases could be done by creating a custom REST API and interfacing this inside of Neos.

**Comment by MR-Alex42** [comment]
These are two different aspects here:
- Integration with external databases via REST API
- A query language inside Neos. 

Examples:
SELECT Slot from RootSlot WHERE Slot.Active.Value = True.
DELETE Slot from RootSlot WHERE Slot.MeshRender.Material.Color,Value = #FF0000.

**Comment by Frooxius** [comment]
It seems that there are actually two different features then? One for integrating with external database support and another is query language for Neos' data model? 

For the latter part LINQ can be utilized, since Neos already has various querying options in its codebase. This would require the full scripting support to compile Linq queries though.

**Comment by Simulacron3** [comment]
I may not be understanding this fully, but I'm specifically interested in working with the Neo4j graph database from Neos vias their REST as a starter. There are well-developed binders for Python and Javascript, and something is being developed for C#. I would guess that basic set of simple queries shouldn't be too difficult to obtain or write in C#. What would a query language in Neos look like?

**Comment by MR-Alex42** [comment]
Yeah, this looks like two different features. However if Neos would be REST enabled this would include the option to query external databases. LINQ would be natural choice for C# based queries of data structures.
